### PR TITLE
Catching Apples C++

### PR DESCRIPTION
**Pull request by HarryWangATX**
_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making the changes.

### CODE REVIEW

File: solutions/usaco-1233.mdx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(N \log N)$
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+CodeSnip{Benq Template}
```
Comment (SansPapyrus683) [inline solutions/usaco-1233.mdx:19]: ...

Comment (HarryWangATX) [reply solutions/usaco-1233.mdx:19]: 🤩

**Comment by SansPapyrus683** [comment]
detemplify and i'll review

**Comment by SansPapyrus683** [comment]
harry just remove the template and i'll review please

**Comment by HarryWangATX** [comment]
sry been little busy lately will remove tonight

**Comment by HarryWangATX** [comment]
@SansPapyrus683 done

File: solutions/usaco-1233.mdx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+
+int N, ans = 0;
+map<int, int> mpp;
+struct item {
```
Comment (SansPapyrus683) [inline solutions/usaco-1233.mdx:25]: rename to `Apple`

File: solutions/usaco-1233.mdx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    int q, t, x, n;
+} p[200005];
+
+bool cmp(const item& x, const item& y) {
```
Comment (SansPapyrus683) [inline solutions/usaco-1233.mdx:29]: could you just make this an operator overload?

File: solutions/usaco-1233.mdx [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+
+int N, ans = 0;
```
Comment (SansPapyrus683) [inline solutions/usaco-1233.mdx:23]: no need to be global

File: solutions/usaco-1233.mdx [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    sort(p, p + N, cmp);
+
+    for(int i = 0; i < N; i++) {
+        if (p[i].q == 2) {
+            mpp[p[i].x + p[i].t] += p[i].n;
+        }
+        else {
+            int n = p[i].n;
+
+            while (n) {
```
Comment (SansPapyrus683) [inline solutions/usaco-1233.mdx:51]: please add some comments

File: solutions/usaco-1233.mdx [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+        return x.q > y.q;
+
+    return x.x - x.t < y.x - y.t;
+}
+
+int main() {
+	cin >> N;
+	for(int i = 0; i < N; i++){
+        cin >> p[i].q >> p[i].t >> p[i].x >> p[i].n;
+	}
```
Comment (SansPapyrus683) [inline solutions/usaco-1233.mdx:40]: sus

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-1233.mdx (hunk 6 above)
Comment (HarryWangATX) [reply solutions/usaco-1233.mdx:40]: oops

Comment (HarryWangATX) [reply solutions/usaco-1233.mdx:40]: hold up thats sus it looks fine on my editor

Comment (HarryWangATX) [reply solutions/usaco-1233.mdx:40]: does vs code not have convert spaces to tabs anymore???

Comment (SansPapyrus683) [reply solutions/usaco-1233.mdx:40]: image
idk man, seems fine on my computer

File: solutions/usaco-1233.mdx (hunk 2 above)
Comment (HarryWangATX) [reply solutions/usaco-1233.mdx:24]: technically can be a cow too

Comment (SansPapyrus683) [reply solutions/usaco-1233.mdx:24]: then rename to `Item` and give some better field names

File: solutions/usaco-1233.mdx [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+		if (x - t == y.x - y.t)
+			return q > y.q;
+
+		return x - t < y.x - y.t;
+	}
+} p[200005];
+
+
+int main() {
+	int N, ans = 0;
```
Comment (SansPapyrus683) [inline solutions/usaco-1233.mdx:38]: declare variables where you actually use them

File: solutions/usaco-1233.mdx [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+	map<int, int> mpp;
```
Comment (SansPapyrus683) [inline solutions/usaco-1233.mdx:39]: here too
also, what does `mpp` store?

File: solutions/usaco-1233.mdx [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+
+
+struct item {
+	int q, t, x, n;
+
+	// expand out inequality into 2D space
```
Comment (SansPapyrus683) [inline solutions/usaco-1233.mdx:27]: what does this mean

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-1233.mdx (hunk 2 above)
Comment (HarryWangATX) [reply solutions/usaco-1233.mdx]: i just used the variable letters stated in the problem

Comment (SansPapyrus683) [reply solutions/usaco-1233.mdx]: at least add a comment next to each

File: solutions/usaco-1233.mdx [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+	int q, t, x, n;
+
+	// cow must arrive before apple hits the number line
+	// let item i be cow and item j be apple
+	// ti <= tj and abs(xi-xj) <= tj-ti can be expanded into the following:
+	// xj-tj <= xi-ti 
+	// xi+ti <= xj+tj
+	// we can represent each item as a point (xi - ti, xi + ti)
+	// for each cow as long as the apple item is to its left and up it is a valid point
+	// therefore we first sort by xi - ti and then greedily select apples based on xi + ti
```
Comment (SansPapyrus683) [inline solutions/usaco-1233.mdx:34]: at this point just take it out into the editorial

**Comment by SansPapyrus683** [comment]
could u submit again & see if it works

**Comment by HarryWangATX** [comment]
it passed

**Review by SansPapyrus683 — APPROVED** [review]
adequate

**Review by envyaims — APPROVED** [review]
omg great

**Pull request merged**